### ISSUE TITLE
cabal check misclassifies -dynamic as debug flag, prevents hackage upload

### ISSUE THREAD

**Issue by nh2** [haskell/cabal#2329]
I'm trying to upload [`call-haskell-from-anything`](https://github.com/nh2/call-haskell-from-anything) to Hackage. It fails with `cabal check`'s

```
* 'ghc-options: -d*' debug flags are not appropriate for a distributed package.
```

because I have `-dynamic`.

`call-haskell-from-anything` doesn't make sense without the `-dynamic` flag, but that isn't really relevant since the core problem is that just checking if the flag starts with `d`:

```haskell
  , check (any ("-d" `isPrefixOf`) all_ghc_options) $
      PackageDistInexcusable $
        "'ghc-options: -d*' debug flags are not appropriate for a distributed package."
```

isn't a good way to classify debug flags.

**Comment by tibbe** [comment]
Cabal version?

**Comment by 23Skidoo** [comment]
This is already fixed in HEAD/1.22 (921921619ff69a7b7d524c6e94dd912859dc51f7). Please open a ticket on the `hackage-server` tracker.

**Comment by nh2** [comment]
Filed in https://github.com/haskell/hackage-server/issues/308 - thanks!
